Server/controllers/ParcelController.js

```diff
@@ -30,49 +30,60 @@ class ParcelController {
 
   // this is to get all parcels by the user
   static async getParcelsByUser(req, res) {
-    const { userId } = req.body;
+    const { userId } = req.user.user.user_id;
     try {
         const response = await parcelmanger.getAllUsersParcelOrder(userId);
         res.status(200).json({
             message: 'got all this users parcels',
-            parcels: response.fields
+            parcels: response.rows
         })
-        console.log('all this users parcels',response.fields)
     }catch(e) {
-        console.log(e)
+        return e;
     }
 }
 
 // this is the controller to get all parcels in the application and it should be accessible by the admin only
 static async getAllParcels (req, res) {
-    console.log('the request object', req.user);
     try {
         const response = await parcelmanger.getAllParcels();
         return res.status(200).json({
-        message: "there was success, all parcels have been fetched",
+        message: "there was success admin, all parcels have been fetched",
         allParcels: response.rows[0]
     })
-    }catch (e) {
-        return res.status(400).json({
-            message: "error in retrieving",
-            e
+    }catch (error) {
+        return res.status(401).json({
+            message: "error in retrieving parcels, you are not authorized",
+            error
         })
     }
 }
 
+// this should enable an admin to get all parcels for a particular user
+static async getAllParcelsBySpecificUser(req, res) {
+    const { uid } = req.params;
+    try {
+        const response = await parcelmanger.getSpecificUsersParcel(uid);
+        res.status(200).json({
+            message: 'got all this users parcels',
+            parcels: response.rows[0]
+        })
+    }catch(e) {
+        return e;
+    }
+}
+
 // this is the controller to update the status of a parcel
 static async updateParcelStatus (req, res) {
     const { pid } = req.params;
     const { newStatus } = req.body;
-    const validStatus = ['in-transit', 'delivered', 'cancelled'].includes(newStatus);
+    const validStatus = ['in-transit', 'delivered'].includes(newStatus);
```

Comment:
For clarification, is `cancelled` not a valid option? I believe the user should be able to cancel the send order.

Reply:
no this is not a route for the user, this is accessible to the admin.
the user has a route that handles the cancel request.

Reply:
ok cool